### PR TITLE
adds new plants, new material reagents

### PR DESCRIPTION
**Pull request by DerFlammenwerfer**
## About The Pull Request
<details>
<summary>
	Adds some new material plants to mutate, reagent versions of several materials that lacked them, and a couple of snowflake chems that will likely be used for future content.
</summary>
<hr>

	
<hr>
</details>

## Changelog
:cl:
add: platinellium now exists, and can be mutated by inserting gold and silver production into plastellium, then blasting it with a somatoray set to evolve. It contains platinum.

tweak: Plastellium now produces plasticide equal to 1/5th of potency (21u max, up from 11)

add: Adds osminellium, which can be obtained by inserting thermite production into platinellium and using a somatoray set to evolve. It contains osmium.

add: Adds lithium nut, which is obtained by mutating peanuts with a somatoray set to mutate, and is a precursor to tritium nut.

add: Adds tritium nut, which is produced by inserting uranium production into lithium nut, and blasting it with evolve rays. Contains tritium, and trace amounts of helium (which currently does nothing, really)

add: Adds heavy water melon, which is mutated from watermelon with mutate rays. Contains deuterium, and is a precursor to metallic melon.

add: Adds metallic melon, which is mutated from heavy water melon that has been given tritium and thermite production and blasted with evolving rays. Contains metallic hydrogen.

add: Adds reagent versions of osmium, mhydrogen, platinum, and tritium.

add: Adds deuterium, aka heavy water. Don't drink more than 30u at a time.

add: Also adds liquid helium, which will be used for other things yet to come.

/:cl:

### CODE REVIEW

File: code/modules/hydroponics/seed_datums.dm [hunk 1]
```diff
@@ -1123,10 +1172,11 @@
 	name = "watermelon"
 	seed_name = "watermelon"
 	display_name = "watermelon vine"
+	greatMutants = list("deuteriumelon")
 	chems = list("nutriment" = list(1,6), "watermelonjuice" = list(10,6))
 
 /datum/seed/watermelon/New()
-	..()
```
Comment (Trilbyspaceclone) [inline code/modules/hydroponics/seed_datums.dm:1129]: Is this intented?

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 2]
```diff
@@ -433,6 +433,14 @@ var/global/chicken_count = 0
 				egg_type = /obj/item/stack/material/plasma
 			if(MATERIAL_URANIUM)
 				egg_type = /obj/item/stack/material/uranium
+			if(MATERIAL_PLATINUM)
+				egg_type = /obj/item/stack/material/uranium
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:437]: It think this is intented to be plat

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 3]
```diff
@@ -433,6 +433,14 @@ var/global/chicken_count = 0
...
+			if(MATERIAL_OSMIUM)
+				egg_type = /obj/item/stack/material/uranium
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:439]: Did you mean osmium?

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 2 above)
Comment (r4d6) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:437]:
```suggestion
				egg_type = /obj/item/stack/material/platinum
```

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 3 above)
Comment (r4d6) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:439]:
```suggestion
				egg_type = /obj/item/stack/material/osmium
```

**Review by r4d6 — COMMENTED** [review]
Just two of the outputs were not set to the correct material type, but otherwise it work fine.

**Comment by LongSleeves** [comment]
nice to see ways to aquire these materials being added.

**Comment by cdb-is-not-good** [comment]
This feels like a straight buff. The material cost of their tech is one of the balancing factors in a lot of the stuff they can produce. They've already got the ability to invalidate some of that via other plants/slimes(i believe?), is pushing further in that direction such a good idea?

**Comment by Dopiotl** [comment]
From my perspective, it's been a staple not to be able to produce rare materials from plants. This is a straight buff, moderated by the time-sink those plants always, always need to go trough before producing anything.

I already felt concerned by the plasteel caps, and asked Risikio to make it so those mutate from a fruit that is only harvested outside the colony, namely in hard to get places such as the hunter's caverns, the greyson gauntlet, the prepper's bunker, etc etc.

I would like to recommend putting all these mutations for material-plants in the same line of "must harvest special weird mushroom or root in very dangerous dungeons of the game.

Being able to time sink alone from shift start just to replace mining doesn't quite sit right by me, and on top of that I find it's just plain bad game design to encourage that.

But, you know, opinions. Oh well

**Comment by r4d6** [comment]
> From my perspective, it's been a staple not to be able to produce rare materials from plants. This is a straight buff, moderated by the time-sink those plants always, always need to go trough before producing anything.
> 
> I already felt concerned by the plasteel caps, and asked Risikio to make it so those mutate from a fruit that is only harvested outside the colony, namely in hard to get places such as the hunter's caverns, the greyson gauntlet, the prepper's bunker, etc etc.
> 
> I would like to recommend putting all these mutations for material-plants in the same line of "must harvest special weird mushroom or root in very dangerous dungeons of the game.
> 
> Being able to time sink alone from shift start just to replace mining doesn't quite sit right by me, and on top of that I find it's just plain bad game design to encourage that.
> 
> But, you know, opinions. Oh well

I just checked the code, and this PR doesn't add any solidification method other than the Robotic Chicken.
Which mean that you need either a Roboticist or CRO to make the chicken before you can start using the plants to make those rare chems.

File: code/modules/hydroponics/seed_datums.dm (hunk 1 above)
Comment (DerFlammenwerfer) [reply code/modules/hydroponics/seed_datums.dm:1129]: No

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 3 above)
Comment (DerFlammenwerfer) [reply code/modules/mob/living/simple_animal/friendly/farm_animals.dm:439]: I did

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 2 above)
Comment (DerFlammenwerfer) [reply code/modules/mob/living/simple_animal/friendly/farm_animals.dm:437]: I did.

**Comment by DerFlammenwerfer** [comment]
> I just checked the code, and this PR doesn't add any solidification method other than the Robotic Chicken. Which mean that you need either a Roboticist or CRO to make the chicken before you can start using the plants to make those rare chems.

This is correct, and intentional, that the robo-chicken is required to solidify the new rare materials.

> I already felt concerned by the plasteel caps, and asked Risikio to make it so those mutate from a fruit that is only harvested outside the colony, namely in hard to get places such as the hunter's caverns, the greyson gauntlet, the prepper's bunker, etc etc.

Of all the plants you can **currently** get, plasteel caps require the most effort, and *none* of the rare material plants I've added here can be produced without first being able to produce somatorays. That's more than just sinking time alone from roundstart. You cannot produce them at all from roundstart without first doing your primary job as a scientist.

File: code/modules/hydroponics/seed_datums.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/modules/hydroponics/seed_datums.dm:1129]: Can you fix it?

**Comment by MSieghardt** [comment]
Soteria is far too self-suficient to allow stuff like this. High tier minerals should not be available through plants.

**Comment by DerFlammenwerfer** [comment]
> Soteria is far too self-suficient to allow stuff like this. High tier minerals should not be available through plants.

It's more to make these particular materials renewable and even then, I almost never need them to begin with except for plasma cells/guns, which I rarely ever use anyway, and other people rarely order. Even then, the durand and phazon both require items that science cannot really produce themselves (magboots and hand-tele respectively), not sure if that applies to the gygax.

Miners will still usually be able to deliver these materials sooner and in greater bulk than science would actually be able to grow their own. 

Platinum *will* be made available through plants for the simple fact that one of the roundstart trade stations buys the greyson tools. This gives science a means to secure some income.

**Comment by r4d6** [comment]
> 

Soteria 100% can make a hand-teleporter.
Granted it require a fuck-ton of Gold 'n stuff to make the artificial bluespace crystal needed, but they can.

**Comment by Yawet330** [comment]
> Soteria is far too self-suficient to allow stuff like this. High tier minerals should not be available through plants.

Agreed. Soteria shouldn't even be the ones with the ((Tiberium processor)), but that's not the point; the main issue currently is that Soteria is far too self-sufficient, and capable of generating near-infinite resources anyway using borgs.

This is even worse when you consider it; because regular botanists can do this.

Its just asking for the economy to be fucked.

**Comment by Risikio** [comment]
Platinum Plant: No. Xenobiology is down the hall. I could've coded in plat production, but ultimately decided not to because plat production then remains something that remains a Xenobio exclusive. 

Other plants: Mostly endorse, though I think the Tritium nut should be a little harder to produce.

**Comment by DerFlammenwerfer** [comment]
> Platinum Plant: No. Xenobiology is down the hall. I could've coded in plat production, but ultimately decided not to because plat production then remains something that remains a Xenobio exclusive.
> 
> Other plants: Mostly endorse, though I think the Tritium nut should be a little harder to produce.

Nobody. Literally no one. Not a single soul does xenobio. Too much RNG involved for too little actual benefit.

**Comment by DerFlammenwerfer** [comment]
> > 
> 
> Soteria 100% can make a hand-teleporter. Granted it require a fuck-ton of Gold 'n stuff to make the artificial bluespace crystal needed, but they can.

Not through the protolathe, that's for damn sure.

**Comment by DerFlammenwerfer** [comment]
> This is even worse when you consider it; because regular botanists can do this.
No they can't. Only gold apples and silverbells can be made by botanists, and that's only if they can source the mutagen themselves. They don't otherwise have access to xenoflora equipment nor floral somatorays.
> Its just asking for the economy to be fucked.
How so? Lonestar can still offer materials in larger quantities far more quickly than xenoflora can produce them. If anything, this would benefit them because science would be more willing and able to make greyson tools for them to sell.

**Comment by r4d6** [comment]
> > > 
> > 
> > 
> > Soteria 100% can make a hand-teleporter. Granted it require a fuck-ton of Gold 'n stuff to make the artificial bluespace crystal needed, but they can.
> 
> Not through the protolathe, that's for damn sure.

_**Anyone**_ can craft a makeshift hand teleporter if they have the mats for it, and it count as a hand-tele for purposes such as building a Phazon.

**Comment by Kazkin** [comment]
While I like the idea, this would be a bad balance change for Soteria, which is having its xenobotany departments ability to produce materials heavily looked at already. As we discussed in the voice call with several people, xenobotany will likely be nerfed in future updates, as such, this one is being closed.

**Comment by DerFlammenwerfer** [comment]
> While I like the idea, this would be a bad balance change for Soteria, which is having its xenobotany departments ability to produce materials heavily looked at already. As we discussed in the voice call with several people, xenobotany will likely be nerfed in future updates, as such, this one is being closed.

As was also discussed in the voice call, and as some of my main opponents agreed, including the guy who made the xenoflora code for material synthesis, Lonestar getting Ameridium machinery, **which they should be and are hard-slated to** in order to prevent Jeanne trying to snipe it again, would make it so that a nerf to xenoflora would no longer be necessary, as Lonestar would have a much more efficient way to provide long-term materials.

**Pull request closed without merging**